ResearchKit/Common/ORKPasscodeStepViewController.m

```diff
@@ -369,7 +369,17 @@ - (void)promptTouchId {
                                                             }]];
                     [strongSelf presentViewController:alert animated:YES completion:nil];
                 } else if (error.code == LAErrorUserCancel) {
-                    [strongSelf makePasscodeViewBecomeFirstResponder];
+
+                    // call becomeFirstResponder here to show the keyboard. dispatch to main queue with
+                    // delay because without it, the transition from the touch ID context back to the app
+                    // inexplicably causes the keyboard to be invisble. It's not hidden, as user can still
+                    // tap keys, but cannot see them
+                    
+                    double delayInSeconds = 0.3;
```

Comment:
@joshbruhin 
Thanks for figuring this out. How did you choose this delay? I seem to recall (fuzzy on this one) that standard animation time for iOS presentation/dismissal is 0.2 seconds?